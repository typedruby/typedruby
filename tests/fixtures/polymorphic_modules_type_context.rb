module Foo::[T]
  def get_t => T; end

  def get_self => :self; end
end

class Bar
  extend Foo::[Integer]
end

def main => nil
  reveal_type(Bar.get_t)
  reveal_type(Bar.get_self)
  nil
end
