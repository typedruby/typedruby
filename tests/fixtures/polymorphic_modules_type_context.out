info: Typechecking...
error: Revealed type is: Integer

        @ __ROOT__/tests/fixtures/polymorphic_modules_type_context.rb:12
     12 |    reveal_type(Bar.get_t)
                         ^^^^^^^^^ expression

error: Revealed type is: Class::[Bar]

        @ __ROOT__/tests/fixtures/polymorphic_modules_type_context.rb:13
     13 |    reveal_type(Bar.get_self)
                         ^^^^^^^^^^^^ expression
